app/screens/edit_server/index.tsx

```diff
@@ -57,7 +63,20 @@ const EditServer = ({closeButtonId, componentId, server, theme}: ServerProps) =>
     }, [componentId]);
 
     useEffect(() => {
-        setButtonDisabled(Boolean(!displayName || displayName === server.displayName));
+        const loadCredentials = async () => {
+            const credentials = await getServerCredentials(server.url);
+            const hasPassword = Boolean(credentials?.preauthSecret);
+
+            // If there's an existing password, show dummy value "keep"
+            const initialValue = hasPassword ? 'keep' : '';
+            setPreauthSecret(initialValue);
+        };
+        loadCredentials();
+    }, [server.url]);
+
+    useEffect(() => {
+        // Allow saving if display name is valid, regardless of changes
+        setButtonDisabled(!displayName);
```

Comment:
You want the save button only when "something has changed", right? So the condition should be something like...
`setButtonDisabled(!displayName || (displayName === server.displayName && !preauthSecretModified)`